gapic/src/main/com/google/gapid/perfetto/views/CounterPanel.java

```diff
@@ -250,5 +257,10 @@ public HoverCard(
       this.size = new Size(leftWidth + HOVER_PADDING + rightLabel + Math.max(minSize.w, maxSize.w),
           Math.max(valueSize.h + avgSize.h, minSize.h + maxSize.h) + HOVER_PADDING);
     }
+
+    public double getTooltipX() {
+      // If the tooltip is smaller than the sample, show it after the sample.
+      return ((size.w < endX - startX) ? mouseX + CURSOR_SIZE / 2 : endX) + HOVER_MARGIN;
```

Comment:
Puzzle: The code to me looks more like: "If the tooltip is smaller than the sample, show it after the mouse; If the tooltip is larger than the sample, show it after the sample.", which is opposite from the above comment.